torch/distributed/elastic/rendezvous/default_rendezvous.py

```diff
@@ -0,0 +1,293 @@
+    """
+
+    _run_id: str
+    _store: Store
+    _backend: RendezvousBackend
+    _min_participants: int
+    _max_participants: int
+    _timeout: RendezvousTimeout
+
+    def __init__(
+        self,
+        run_id: str,
+        store: Store,
+        backend: RendezvousBackend,
+        min_participants: int,
+        max_participants: int,
+        timeout: Optional[RendezvousTimeout] = None,
+    ) -> None:
+        if not run_id:
+            raise ValueError("The run id must be a non-empty string.")
+
+        if min_participants < 1:
+            raise ValueError(
+                f"The minimum number of participants ({min_participants}) must be greater than "
+                "zero."
+            )
+
+        if max_participants < min_participants:
+            raise ValueError(
+                f"The maximum number of participants ({max_participants}) must be greater than or "
+                f"equal to the minimum number of participants ({min_participants})."
+            )
+
+        self._run_id = run_id
+
+        self._store = store
+        self._backend = backend
+
+        self._min_participants = min_participants
+        self._max_participants = max_participants
+
+        self._timeout = timeout or RendezvousTimeout()
+
+    @property
+    def run_id(self) -> str:
+        """Gets the run id of the rendezvous."""
+        return self._run_id
+
+    @property
+    def store(self) -> Store:
+        """Gets the C10d store returned as part of the rendezvous."""
+        return self._store
+
+    @property
+    def backend(self) -> RendezvousBackend:
+        """Gets the backend used to hold the rendezvous state."""
+        return self._backend
+
+    @property
+    def min_participants(self) -> int:
+        """Gets the minimum number of participants to admit to the rendezvous."""
+        return self._min_participants
+
+    @property
+    def max_participants(self) -> int:
+        """Gets the maximum number of participants to admit to the rendezvous."""
+        return self._max_participants
+
+    @property
+    def timeout(self) -> RendezvousTimeout:
+        """Gets the timeout configuration of the rendezvous."""
+        return self._timeout
+
+    def get_backend(self) -> str:
+        """See base class."""
+        return self._backend.name
+
+    def next_rendezvous(self) -> Tuple[Store, int, int]:
+        """See base class."""
+        raise NotImplementedError()
+
+    def is_closed(self) -> bool:
+        """See base class."""
+        raise NotImplementedError()
+
+    def set_closed(self) -> None:
+        """See base class."""
+        raise NotImplementedError()
+
+    def num_nodes_waiting(self) -> int:
+        """See base class."""
+        raise NotImplementedError()
+
+    def get_run_id(self) -> str:
+        """See base class."""
+        return self._run_id
+
+    def shutdown(self) -> bool:
+        """See base class."""
+        raise NotImplementedError()
```

Comment:
I might be missing something here, but the PR says the bulk of the rendezvous logic would be in this class. But its not implemented here and I don't see it in the other two PRs on top of it (C10dRendezvous and EtcRendezvous backends). Is there another PR in the works that adds this functionality?

Reply:
Yes, I briefly mentioned it in the description as "as a stub implementation". It is actually the core piece of this whole task and will have its own PR by tomorrow.